### DISCUSSION TITLE
make plugin_local  compile error

### DISCUSSION THREAD

**Discussion by jcy1001** [alibaba/ilogtail#683] [Help] [answered]
make all success
but make plugin_local  compile error

root@DESKTOP-C48UTU1:/home/jcy/ilogtail# go mod tidy 
root@DESKTOP-C48UTU1:/home/jcy/ilogtail# make plugin_local
./scripts/plugin_build.sh mod c-shared output
Linux
x86_64
# github.com/coreos/go-systemd/sdjournal
/root/go/pkg/mod/github.com/coreos/go-systemd@v0.0.0-20190719114852-fd7a80b32e1f/sdjournal/journal.go:27:11: fatal error: systemd/sd-journal.h: No such file or directory
   27 | // #include <systemd/sd-journal.h>
      |           ^~~~~~~~~~~~~~~~~~~~~~
compilation terminated.
make: *** [Makefile:118: plugin_local] Error 2

**Comment by messixukejia** [comment]
操作的环境是：root@DESKTOP-C48UTU1，从这里看是个人的主机，并不是在开发镜像里。具体参见：https://ilogtail.gitbook.io/ilogtail-docs/developer-guide/development-environment

如果确实需要本地操作，上面问题可以通过systemd-devel: sudo dnf install systemd-devel。但是后续可能还会遇到其他兼容性问题。这些都是一些很基本的兼容性问题，[网上搜一下](https://www.google.com/search?q=fatal+error%3A+systemd%2Fsd-journal.h&oq=fatal+error%3A+systemd%2Fsd-journal.h&aqs=chrome..69i57j69i61.222j0j7&sourceid=chrome&ie=UTF-8)会有很多解决办法。

如果有开发需求，建议参与下双周会进行交流。一方面可以反馈下诉求，另一方面也可以将自己的经验分享给更多人。

**Comment by jcy1001** [comment, marked as answer]
![image](https://user-images.githubusercontent.com/24226340/222619729-fe72e933-cbe9-4fdf-910b-be6129404434.png)

**Reply by jcy1001** [reply]
when I make  success ，i found there is no  plugin_base.so ,so I make plugin_main , error occurs

**Reply by yyuuttaaoo** [reply]
After `make plugin`，plugin_base.so should be in output dir.

According to your snapshot, it is likely that you machine's arch is ARM. You can first `make all` and then copy `output/libPluginAdapter.so` to overwrite `pkg/logtail/libPluginAdapter.so`. After that, you can `make plugin_main`. However is won't produce `plugin_base.so`.
